### PR TITLE
LPD-21217 Implement disabling sorting control & test v2

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPD-6704 Disable sorting of hierarchical fields](https://issues.liferay.com/browse/LPD-6704)
- This is a followup on https://github.com/liferay-frontend/liferay-portal/pull/4089
- Related to [LPD-20684 Flaky fields tests](https://liferay.atlassian.net/browse/LPD-20684)

### What is the goal of this PR?

Compared to the previous version, we are:
- Simplifying solution logic, by moving much of it to backend. This way it works for both post-FF treeview modal and pre-FF list modal.
- Fixing the default sortable values, after fields are added for the first time. They are now `true`, unless the field type is object. 
- Rewriting tests. There is some overlap between this solution and fixing [LPD-20684 Flaky fields tests](https://liferay.atlassian.net/browse/LPD-20684), related to API data creation. I ended up copying some code from `fields.spec.ts`, as I moved the code to `visualizationModes.spec.ts`. But, there is still a lot more to do: no site isolation, usage of old UI with no visualization tabs, and url error on fragment publish. And it was convenient to reuse `FDSFields` as sample date. We can complete all of this in the bug.

### What does it look like?

image

image

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx [hunk 1]
```diff
@@ -159,14 +159,17 @@ const SaveFDSFieldsModalContent = ({
 	const saveFDSFields = async () => {
 		setSaveButtonDisabled(true);
 
-		const creationData: Array<{name: string; type: string}> = [];
+		const creationData: Array<{
+			name: string;
+			type: string;
+		}> = [];
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx:165]: Both versions are fine :open_mouth:

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/action/SaveFDSFieldsMVCResourceCommand.java [hunk 2]
```diff
@@ -84,9 +87,9 @@ protected void doTransactionalCommand(
 				).put(
 					"renderer", "default"
 				).put(
-					"sortable", true
+					"sortable", !StringUtil.equals(type, "object")
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/action/SaveFDSFieldsMVCResourceCommand.java:90]: Nice!

**Review by juanjofgliferay — APPROVED** [review]
LGTM :rocket:

**Comment by juanjofgliferay** [comment]
@markocikos Regarding the overlaping with `fields.spec.ts`, does it make sense to move all tests to the same file -- `visualizationModes.spec.ts`?

**Comment by markocikos** [comment]
> Regarding the overlaping with fields.spec.ts, does it make sense to move all tests to the same file -- visualizationModes.spec.ts?

@juanjofgliferay Yep, definitely. I was thinking to do it in this PR, but PR is already big enough. When we added `fields.spec.ts`, there was no visualization mode tab. It explicitly works with old UI  (`'LPD-10735': false`).

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/view/visualization_modes/VisualizationModesPage.ts [hunk 3]
```diff
@@ -30,6 +44,25 @@ export class VisualizationModesPage {
...
+	async addRootField(field: string) {
+		await this.addFieldsDialog.fields
+			.getByText(field, {exact: true})
+			.click();
+	}
+
+	async cancelAddFieldsModal() {
+		await this.addFieldsDialog.cancelButton.click();
+	}
+
```
Comment (dsanz) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/view/visualization_modes/VisualizationModesPage.ts:65]: Nice! functions encapsulating the interactions!

**Review by dsanz — APPROVED** [review]
LGTM ;)

**Comment by markocikos** [comment]
ci:report:2558209623

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:test:sf

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
No related Playwright errors, based on https://github.com/liferay-frontend/liferay-portal/pull/4108#issuecomment-2037224440

**Comment by markocikos** [comment]
ci:forward:force

**Pull request closed without merging**